pkg/sql/testdata/logic_test/builtin_function

```diff
@@ -1534,22 +1534,25 @@ six-one-seven
 minus-two
 
 # Do some basic sanity checking of the variadic hash functions.
-query BBBBBB
+query BBBBBBBBB
 SELECT
   sha512('1') = sha512('1'),
   sha512('1') = sha512('2'),
   sha512('1', '2') = sha512('1', '2'),
   sha512('1', '2') = sha512('2', '1'),
   sha512('1', '2') = sha512('12'),
-  sha512('1', '2') = sha512('21')
+  sha512('1', '2') = sha512('21'),
+  sha512('bar') = sha512(b'bar':::bytes),
+  sha512(b'bar'::bytes) = sha512(b'bar':::bytes),
+  sha512(b'bar'::bytes) = sha512('bar')
 ----
-true false true false true false
+true false true false true false true true true
 
-# The hash functions should be stable, so verify that the  following hashes
+# The hash functions should be stable, so verify that the following hashes
 # don't change.
 query T
 SELECT i FROM (VALUES
-  (sha512(NULL)),
+  (sha512(NULL::string)),
```

Comment:
We could avoid this via a `preferredOverload` hack, but it doesn't seem worth it -- IIRC @knz and @nvanbenschoten would rather avoid using `preferredOverload` except when absolutely forced to for pg compat.